ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java

```diff
@@ -0,0 +1,416 @@
+import org.apache.ignite.cache.QueryIndexType;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.configuration.BinaryConfiguration;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgnitionEx;
+import org.apache.ignite.internal.binary.BinaryMarshaller;
+import org.apache.ignite.logger.slf4j.Slf4jLogger;
+import org.hibernate.ogm.datastore.ignite.IgniteConfigurationBuilder;
+
+/**
+ * Ignite cache configuration for tests
+ *
+ * @author Victor Kadachigov
+ */
+public class IgniteTestConfigurationBuilder implements IgniteConfigurationBuilder {
+
+	@Override
+	public IgniteConfiguration build() {
+		//disable check for new versions
+		System.setProperty( IgniteSystemProperties.IGNITE_UPDATE_NOTIFIER, Boolean.FALSE.toString() );
+		
+		IgniteConfiguration config = null;
+
+		try {
+			//config = loadFromResource( "ignite-config.xml" );
+			config = createConfig();
+		}
+		catch (Exception ex) {
+			throw new RuntimeException( ex );
+		}
+
+		return config;
+	}
+
+	private IgniteConfiguration loadFromResource(String resourceName) throws IgniteCheckedException {
+		URL url = getClass().getClassLoader().getResource( resourceName );
+		return IgnitionEx.loadConfiguration( url ).get1();
+	}
+
+	private IgniteConfiguration createConfig() {
+		IgniteConfiguration config = new IgniteConfiguration();
+
+		config.setGridName( "OgmTestGrid" );
+		config.setLocalHost( "127.0.0.1" );
+		config.setPeerClassLoadingEnabled( true );
+		config.setClientMode( false );
+		config.setMarshaller( new BinaryMarshaller() ) ;
+		BinaryConfiguration binaryConfiguration = new BinaryConfiguration();
+		binaryConfiguration.setCompactFooter( false );		// it is necessary only for embedded collections (@ElementCollection)
+		config.setBinaryConfiguration( binaryConfiguration );
+		config.setGridLogger( new Slf4jLogger() );
+		config.setPublicThreadPoolSize( 2 );
+
+		List<CacheConfiguration> cacheConfig = new ArrayList<>();
+
+// EmbeddableIdTest
+		cacheConfig.add( createCacheConfig( "SingleBoardComputer" ).withForceQueryEntity().build() );
+// SequenceIdGeneratorTest
+		cacheConfig.add( simpleCacheConfig( "Song", Long.class ) );
+		cacheConfig.add( simpleCacheConfig( "Actor", Long.class ) );
+// JPAResourceLocalTest
+		cacheConfig.add( simpleCacheConfig( "Poem" ) );
+// ManyToManyTest
+		cacheConfig.add(
+				createCacheConfig( "Car" )
+					.withKeyType( "CarId" )
+					.build()
+		);
+		cacheConfig.add(
+				createCacheConfig( "Tire" )
+					.withKeyType( "TireId" )
+					.build()
+		);
+		cacheConfig.add(
+				createCacheConfig( "Car_Tire" )
+					.appendIndex( "cars_carId_maker", String.class )
+					.appendIndex( "cars_carId_model", String.class )
+					.appendIndex( "tires_tireId_maker", String.class )
+					.appendIndex( "tires_tireId_model", String.class )
+					.build()
+		);
```

Comment:
It's not really about IgniteTestConfigurationBuilder, I'm wondering if the configuration of the cache can be created at start up by OGM using the information from the existing annotation. This is the approach we use in OGM, unless it is no feasable for some reasons. And that's also the idea behind the SchemaDefiner interface. For example, we already know that those columns are identifeirs of an association and therefore we could create an index when we create the cache.

This way the user does not have to provide a configuration for the caches unless he needs some more customization that we don't provide out of the box or at elast he nedds to provide less information.

For example. in Neo4j we add unique constraints when a property is mapped using `@Column(unique=true)`. Not all datastores allow this and I'm wondering if it's feasable with Ignite.

I'm not saying that you have to do everuthing now, though